src/coreclr/jit/emit.cpp

```diff
@@ -4102,6 +4126,200 @@ void emitter::emitDispCommentForHandle(size_t handle, GenTreeFlags flag)
+
+            if ((targetGroup != nullptr) && (jmpGroup->igNext == targetGroup))
+            {
+                // the last instruction in the group is the jmp we're looking for
+                // and it jumps to the next instruction group so we don't need it
+                CLANG_FORMAT_COMMENT_ANCHOR
+
+#ifdef DEBUG
+                unsigned instructionCount = jmpGroup->igInsCnt;
+                assert(instructionCount > 0);
+                instrDesc* id = nullptr;
+                {
+                    BYTE* dataPtr = jmpGroup->igData;
+                    while (instructionCount > 0)
+                    {
+                        id = (instrDesc*)dataPtr;
+                        dataPtr += emitSizeOfInsDsc(id);
+                        instructionCount -= 1;
+                    }
+                }
+                assert(id != nullptr);
+                if (jmp != id)
+                {
+                    printf("jmp != id, dumping context information\n");
+                    printf("method: %s\n", emitComp->impInlineRoot()->info.compMethodName);
+                    printf("  jmp: %x3: ", jmp->idDebugOnlyInfo()->idNum);
+                    emitDispIns(jmp, false, true, false, 0, nullptr, 0, jmpGroup);
+                    printf("  id: %x3: ", id->idDebugOnlyInfo()->idNum);
+                    emitDispIns(id, false, true, false, 0, nullptr, 0, jmpGroup);
+                    printf("jump group:\n");
+                    emitDispIG(jmpGroup, nullptr, true);
+                    printf("target group:\n");
+                    emitDispIG(targetGroup, nullptr, false);
+                }
+                assert(jmp == id);
+#endif // DEBUG
+
+                JITDUMP("IG%02u IN%04x is the last instruction in the group and jumps to the next instruction group "
+                        "IG%02u %s, removing.\n",
+                        jmpGroup->igNum, jmp->idDebugOnlyInfo()->idNum, targetGroup->igNum,
+                        emitLabelString(targetGroup));
+
+                // Unlink the jump from emitJumpList while keeping the previousJmp the same.
+                if (previousJmp != nullptr)
+                {
+                    previousJmp->idjNext = jmp->idjNext;
+                    if (jmp == emitJumpLast)
+                    {
+                        emitJumpLast = previousJmp;
+                    }
+                }
+                else
+                {
+                    assert(jmp == emitJumpList);
+                    emitJumpList = jmp->idjNext;
+                }
+
+                UNATIVE_OFFSET codeSize = jmp->idCodeSize();
+                jmp->idCodeSize(0);
+
+                jmpGroup->igSize -= (unsigned short)codeSize;
+                jmpGroup->igFlags |= IGF_UPD_ISZ;
+
+                emitTotalCodeSize -= codeSize;
+                totalRemovedSize += codeSize;
+#if DEBUG
+                totalRemovedCount += 1;
```

Comment:
Honestly, I don't see much value in having `totalRemovedCount`. You could just add a `JITDUMP` here to say that you have removed a jump that targets XYZ IG and reduced size so far is XXX bytes.

Reply:
Removed. 

I've always found that it's best to have helpful debugging output and having a count of instructions removed makes it just a little bit easier to see if there's a problem with the jump list relinking.